src/content/marina/marinaBroker.ts

```diff
@@ -326,6 +328,102 @@ export default class MarinaBroker extends Broker {
           return successMsg();
         }
 
+        case Marina.prototype.broadcastTransaction.name: {
+          this.checkHostnameAuthorization(state);
+          const network = selectNetwork(state);
+          const [signedTxHex] = params as [string];
+
+          // broadcast tx
+          const txid = await broadcastTx(selectEsploraURL(state), signedTxHex);
+          if (!txid) throw new Error('something went wrong with the tx broadcasting');
+
+          // will need tx to lock selected utxos and credit unconfirmed utxos
+          const tx = Transaction.fromHex(signedTxHex);
+
+          // lock selected utxos used in this transaction:
+          // - get all coins from marina
+          // - iterate over all transaction inputs and check if is a coin of ours
+          // - if any, lock selected utxos
+
+          // get all coins from marina
+          const coins = selectUtxos(MainAccountID)(state);
```

Comment:
Do we need a selector to retrieve all possible accounts here cc/ @louisinger

Reply:
What about marina.getCoins?

Reply:
You can use `selectUtxos(selectAllAcountsIDs(state)...)(state)`

Reply:
Fixed by 33dafc3